### PR TITLE
Move upsert to shell command activity

### PR DESCRIPTION
**Pull request by sb2nov**
@tpcstld PTAL

### CODE REVIEW

File: dataduct/steps/create_load_redshift.py [hunk 1]
```diff
@@ -41,7 +40,7 @@ def __init__(self, id, table_definition, input_node=None,
             script_arguments = list()
 
         script_arguments.extend([
-            '--table_definition=%s' % table_exists_script.sql(),
+            '--table_definition=%s' % table.sql().sql(),
```
Comment (tpcstld) [inline dataduct/steps/create_load_redshift.py]: Can you explain why you need two `.sql()`s here?

File: dataduct/database/view.py [hunk 2]
```diff
@@ -46,3 +46,15 @@ def drop_script(self):
...
         return SqlScript('DROP VIEW IF EXISTS %s CASCADE' % self.full_name)
+
+    def check_not_exists_script(self):
+        """Sql script to create statement if the table exists or not
+        """
+        return SqlScript("""
+            SELECT NOT EXISTS(
+                SELECT 1
+                FROM information_schema.views
+                WHERE table_schema = '%s'
```
Comment (tpcstld) [inline dataduct/database/view.py]: should this be `view_schema` and `view_name`?

File: dataduct/steps/scripts/sql_runner.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
```
Comment (tpcstld) [inline dataduct/steps/scripts/sql_runner.py]: No space

File: dataduct/steps/scripts/sql_runner.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
```
Comment (tpcstld) [inline dataduct/steps/scripts/sql_runner.py]: Rename to upsert_runner.

File: dataduct/steps/scripts/sql_runner.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                                      connection).loc[0][0]
+
+    cursor = connection.cursor()
+    # Create table in redshift, this is safe due to the if exists condition
+    if table_not_exists:
+        cursor.execute(table.create_script().sql())
+
+    # Load data into redshift with upsert query
+    cursor.execute(args.sql)
+    cursor.execute('COMMIT')
```
Comment (tpcstld) [inline dataduct/steps/scripts/sql_runner.py]: Does this not need a corresponding `BEGIN;`?

File: dataduct/steps/create_load_redshift.py (hunk 1 above)
Comment (sb2nov) [inline dataduct/steps/create_load_redshift.py]: table.sql() returns a sql statement and then the next sql converts it to a string.

File: dataduct/database/view.py (hunk 2 above)
Comment (sb2nov) [inline dataduct/database/view.py]: no 
```
coursera=> \d information_schema.views
                  View "information_schema.views"
       Column       |               Type                | Modifiers
--------------------+-----------------------------------+-----------
 table_catalog      | information_schema.sql_identifier |
 table_schema       | information_schema.sql_identifier |
 table_name         | information_schema.sql_identifier |
 view_definition    | information_schema.character_data |
 check_option       | information_schema.character_data |
 is_updatable       | information_schema.character_data |
 is_insertable_into | information_schema.character_data |
```

File: dataduct/steps/scripts/sql_runner.py (hunk 4 above)
Comment (sb2nov) [inline dataduct/steps/scripts/sql_runner.py]: no as this is shared across multiple sql abstractions that create and run some sql

File: dataduct/steps/scripts/sql_runner.py (hunk 5 above)
Comment (sb2nov) [inline dataduct/steps/scripts/sql_runner.py]: no cursor executes stuff as a transaction already

File: dataduct/steps/scripts/sql_runner.py (hunk 3 above)
Comment (sb2nov) [inline dataduct/steps/scripts/sql_runner.py]: Another diff as wasted commit

**Comment by tpcstld** [comment]
LGTM

**Pull request merged**